server/channels/app/license.go

```diff
@@ -18,10 +18,6 @@ func (ch *Channels) License() *model.License {
 }
 
 func (ch *Channels) RequestTrialLicenseWithExtraFields(requesterID string, trialRequest *model.TrialLicenseRequest) *model.AppError {
-	if *ch.srv.platform.Config().ExperimentalSettings.RestrictSystemAdmin {
-		return model.NewAppError("RequestTrialLicense", "api.restricted_system_admin", nil, "", http.StatusForbidden)
-	}
-
```

Comment:
We don't need to double check this here: we already do the right thing in the API and plugin layers.